entity-server-api/src/main/java/org/terracotta/entity/ServiceProvider.java

```diff
@@ -22,46 +22,62 @@
 
 
 /**
- * Service Provider which are provided at the platform level.  A service provider is responsible for instantiating the
- * services to interact with its underlying implementation in response to platform requests.  While the services are
- * generally used by entities, they are also used by the platform, itself.
+ * A service provider is responsible for instantiating the services to interact with its underlying implementation in response
+ *  to platform requests.  While the services are generally used by entities, they are also used by the platform, itself.
+ * 
+ * Note that there are no threading restrictions on how the methods in this interface will be called.  This means that any
+ *  implementation which has shared state must ensure that it can manage this state in a thread safe way (subject to whatever
+ *  thread safety its own state requires).
  */
 public interface ServiceProvider {
-
   /**
-   * The platform configuration based on which the Service provider can choose to initialize itself.
-   *
-   * @param configuration platform configuration
-   * @return true if provider has successfully been initialized with the provided configuration
+   * The platform configuration based on which the receiver can choose to initialize itself.
+   * This is called once on each provider instance, at server start-up time, before any other methods have been called on the
+   *  instance.
+   * Note that it is possible for there to be multiple instances of the same ServiceProvider if the server had multiple
+   *  configuration stanzas handled by the same ServiceProvider class.
+   * 
+   * @param configuration The configuration for this ServiceProvider instance.
+   * @return True if provider has successfully been initialized with the provided configuration, false if this instance
+   *  shouldn't be registered for the given configuration (it will be discarded).
    */
   boolean initialize(ServiceProviderConfiguration configuration);
 
   /**
-   * Get an instance of service from the provider.
-   *
-   * @param consumerID The unique ID used to name-space the returned service
-   * @param configuration Service configuration which is to be used
-   * @return service instance
+   * Get an instance of a service from the provider.
+   * Note that this method can be called any number of times, over the life of a ServiceProvider instance, potentially
+   *  concurrently or multiple times for the same consumerID.  If the implementation wishes to intern, or otherwise track,
+   *  the instances it returns, it will need to ensure its approach is robust against concurrent calls.
+   * Note that the consumerID is a unique number for each entity instance (consumerID 0 represents the platform, itself) which
```

Comment:
Good call.  I wasn't sure if this is the right place for it but it probably is (and we can move it later, if need be).

I will also add a note that all consumerID values are non-negative.